src/platform/linux_usbfs/device.rs

```diff
@@ -54,8 +56,105 @@ impl LinuxDevice {
+
+        // NOTE: must be called from the same thread that gave us the FD on android, otherwise this fails
+        let proc_path = PathBuf::from(format!("/proc/self/fd/{}", fd.as_raw_fd()));
+
+        debug!("Reading link: {proc_path:?}");
+        let fd_path = std::fs::read_link(&proc_path)?;
+
+        // TODO: should we switch to regex or regex-lite for this?
+        let prefix = "/dev/bus/usb/";
+        let Ok(dev_num_bus_num) = fd_path.strip_prefix(prefix) else {
+            error!(
+                "Usb device path does not start with required prefix `{prefix}`: {}",
+                fd_path.to_string_lossy()
+            );
+            return Err(ErrorKind::Other.into());
+        };
+        let Some(dev_num_bus_num) = dev_num_bus_num.to_str() else {
+            error!("End of usb device path is not UTF-8!: {dev_num_bus_num:?}",);
+            return Err(ErrorKind::Other.into());
+        };
+
+        let mut split = dev_num_bus_num.split("/");
+        let Some(bus_str) = split.next() else {
+            error!("Failed to split usbfs device path: {}", dev_num_bus_num);
+            return Err(ErrorKind::Other.into());
+        };
+        let Some(dev_str) = split.next() else {
+            error!("Failed to split usbfs device path: {}", dev_num_bus_num);
+            return Err(ErrorKind::Other.into());
+        };
+
+        let Ok(busnum) = bus_str.parse::<u8>() else {
+            error!("Usb bus number failed to parse: {bus_str}",);
+            return Err(ErrorKind::Other.into());
+        };
+        let Ok(devnum) = dev_str.parse::<u8>() else {
+            error!("Usb device number failed to parse: {dev_str}",);
+            return Err(ErrorKind::Other.into());
+        };
+
+        const LIBUSB_REQUEST_GET_CONFIGURATION: u8 = 0x08;
+
+        let mut dst = [0u8; 1];
+
+        let control = Control {
+            control_type: ControlType::Standard,
+            recipient: Recipient::Device,
+            request: LIBUSB_REQUEST_GET_CONFIGURATION,
+            value: 0,
+            index: 0,
+        };
+
+        // TODO: what to do about blocking here?
+        let r = usbfs::control(
+            &fd,
+            usbfs::CtrlTransfer {
+                bRequestType: control.request_type(Direction::In),
+                bRequest: control.request,
+                wValue: control.value,
+                wIndex: control.index,
+                wLength: dst.len() as u16,
+                timeout: Duration::from_millis(50)
+                    .as_millis()
+                    .try_into()
+                    .expect("timeout must fit in u32 ms"),
+                data: &mut dst[0] as *mut u8 as *mut c_void,
+            },
+        );
+
+        // TODO: Could add some more logic here to support buggy devices, similar to what libusb does
+        // See: https://github.com/libusb/libusb/blob/467b6a8896daea3d104958bf0887312c5d14d150/libusb/os/linux_usbfs.c#L865
```

Comment:
I wouldn't be surprised if it's common to not implement `GET_CONFIGURATION` because it doesn't seem like it would be commonly used.

If there's only one configuration, which will be the case for the vast majority of devices, it seems pretty safe to assume that's the active configuration.